### ISSUE TITLE
file() or file.text() not working?

### ISSUE THREAD

**Issue by SaifAshrafHelmy** [oven-sh/bun#5960]
### What version of Bun is running?

1.0.3

### What platform is your computer?

Microsoft Windows NT 10.0.19045.0 x64 -- using WSL version: 1.2.5.0

### What steps can reproduce the bug?
```js
const file = Bun.file("myfile.txt");
console.log(file)

const file2 = Bun.file("wrongfilename.txt");
console.log(file2)




console.log(await file.text())


console.log("Hello?");

```

PLEASE NOTICE THAT THE MAIN ISSUE HERE IS THAT :
THE CONTENT OF ACTUAL EXISTING FILE "myfile.txt"  IS NOT BEING READ IN await file.txt
 [video ref](https://youtu.be/GU-CecSn6uY)

### What is the expected behavior?

1- contents of the myfile.txt should be printed to console, but they don't

2- console.log(fil2) should give error or something, but I get 
FileRef ("wrongfilename.txt") {   
  type: "text/plain;charset=utf-8"
}
although this file doesn't exist

3-console.log("Hello?") should print Hello? but it doesn't, none of the code is working after the file.text() is called

### What do you see instead?

FileRef ("myfile.txt") {
  type: "text/plain;charset=utf-8"
}
FileRef ("wrongfilename.txt") {
  type: "text/plain;charset=utf-8"
}

I don't see the contents of myfile.txt (which is a simple Hello Bun text,
and I see FileRef for wrongfilename.txt, although the file doesn't exist

**Comment by jimmywarting** [comment]
Have been playing around with `Bun.file()` myself, and i encounter pretty much the same weird behavior myself.
`Bun.file(path)` works on any path you throw at it. Even file paths that do not exist. 
it's not until you try read the content of it that it will throw an error:
```js
Bun.file('dont-exist.dat').text().then(console.log)

ENOENT: No such file or directory
   path: "dont-exist.dat"
 syscall: "open"
   errno: -2

Promise {
  <rejected> [ENOENT: No such file or directory] {
    code: 'ENOENT',
    path: 'dont-exist.dat',
    syscall: 'open',
    errno: -2
  }
}
```
i think it should throw a DomError-NotFoundError already at `Bun.file('dont-exist.dat')` if it can't locate the file. Not afterwards.

**Comment by melroy89** [comment]
@SaifAshrafHelmy Try to use code blocks using ``` around your code, for easy readability.

**Comment by Electroid** [comment]
This is how the `Bun.file` API is designed, `Bun.file` will always return a File object. It will only throw if you try to do an operation, such as: `Bun.file(...).text()`. You can also check `Bun.file(...).exists()`

**Comment by jimmywarting** [comment]
> Bun.file will always return a File object

here i will disagree with you... it dose actually return a Blob object. I don't think you should be confusing it with the `File` class

`exists()` seems to be embedded in `Blob.prototype.exists` so dose `name`, `json`, `formData`, and `lastModified`
- `Blob.prototype.name`
- `Blob.prototype.lastModified`
- `Bun.file(path).constructor` returns Blob
- `new File()` also returns a blob. name is undefined.

I can appreciate the added functionality

But i don't think you should be polluting things onto the `Blob` class that dose not belong there...
it's unexpected, abnormal and not spec compatible. i think that if you want to see this thing being added onto `Blob` or `File` class then it should go trough via https://github.com/w3c/FileAPI/issues first.

I think it would be better if you instead extended `File` instead

```js
class File extends Blob { ... }
class BunFile extends File { 
  json() { ... }
  exist() { ... }
  writer() { ... }
  formData() { ... }
}
```

**Comment by jimmywarting** [comment]
I genially feel like extending native core classes are a bad idea... even things such as `class A extends Array { ... }` is a bad idea.
just look at NodeJS `Buffer.prototype.slice` fiasco. `Uint8array` have slice too. and overriding it was a misstake. 
`.subarray()` should have been used instead.
but even so they do also override `subarray` too and do still return a `Buffer` instance instead of a plain `Uint8Array`

Extending builtin native host object is like using the old bad historical `prototype.js` since there can be a chance of collisions if host object do decide to add some new method to any existing thing.

**Comment by jimmywarting** [comment]
> This is how the Bun.file API is designed

i wonder what the design goal was...

**Comment by SaifAshrafHelmy** [comment]
> This is how the `Bun.file` API is designed, `Bun.file` will always return a File object. It will only throw if you try to do an operation, such as: `Bun.file(...).text()`. You can also check `Bun.file(...).exists()`

Although this does not seem a good thing to me, but anyway, 

What about the actual bug I mentioned? The actual file is not being read.. why did you close the issue?

**Comment by Electroid** [comment]
> I think it would be better if you instead extended File instead

This is an oversight. I'll make another issue to address.

**Comment by Electroid** [comment]
> The actual file is not being read.. why did you close the issue?

@SaifAshrafHelmy Could you try the following code, what's the output?

```js
console.log(typeof await Bun.file("myfile.txt").text());
```

**Comment by SaifAshrafHelmy** [comment]
> > The actual file is not being read.. why did you close the issue?
> 
> @SaifAshrafHelmy Could you try the following code, what's the output?
> 
> ```js
> console.log(typeof await Bun.file("myfile.txt").text());
> ```

I also get no output at all, and also, the code after it doesn't run.. 

So this code below, when I run it with bun run index.ts, give no output at all
```js
console.log(typeof await Bun.file("myfile.txt").text())
console.log("Hello? Can you write me?");

```

**Comment by SaifAshrafHelmy** [comment]
here's a video record because this seems very weird for me too..
https://youtu.be/GU-CecSn6uY

**Comment by melroy89** [comment]
> here's a video record because this seems very weird for me too.. https://youtu.be/GU-CecSn6uY

it crashes?

I see you are using Windows. This is what happens under Linux on my system:

![it_works_here](https://github.com/oven-sh/bun/assets/628926/9b19512d-c080-4e17-b6ed-d3e87ca9eab5)

**Comment by SaifAshrafHelmy** [comment]
> > here's a video record because this seems very weird for me too.. https://youtu.be/GU-CecSn6uY
> 
> it crashes?
> 
> I see you are using Windows. This is what happens under Linux on my system:
> 
> ![it_works_here](https://github.com/oven-sh/bun/assets/628926/9b19512d-c080-4e17-b6ed-d3e87ca9eab5)
> 

It doesn't crash.. just doesn't output anything? And whatever code I put after it doesn't get executed

**Comment by melroy89** [comment]
>  And whatever code I put after it doesn't get executed

Sounds like a crash to me.. 😕 Also because the application exists as well. Since you are using WSL, can you try to run: `echo $?` in the terminal DIRECTLY after you executed your bun command. And share the output of this.

**Comment by SaifAshrafHelmy** [comment]
> > And whatever code I put after it doesn't get executed
> 
> Sounds like a crash to me.. 😕 Also because the application exists as well. Since you are using WSL, can you try to run: `echo $?` in the terminal DIRECTLY after you executed your bun command. And share the output of this.

yeah maybe you're right..
the exit code is I get is 1

**Comment by melroy89** [comment]
> the exit code is I get is 1

Yea, so this needs some further debugging, but your app is crashing. Maybe try a try/catch or something.. And log the error (if any)? Or try to profile the code.

**Comment by SaifAshrafHelmy** [comment]
nope, try and catch not catching anything at all.. and IDK how to profile the code or even debug it.. the docs only explain how to debug web servers... so, I guess I'll stick to my beloved nodejs lol

**Comment by melroy89** [comment]
@robobun you removed the `crash` label again?

@SaifAshrafHelmy YOu can use the WebKit inspector to enable debugging your code use the `--inspect` flag. See: https://bun.sh/guides/runtime/web-debugger

**Comment by Hit-And-Run-5677** [comment]
Hello I am also facing the same error on the same platform specs. But really want to fix it and use Bun in my next projects. One peculiar thing about that error is- neither BunFile.text() works nor does BunFile.arrayBuffer() but in new Response(Bun.file(...)) works like a charm in Bun.serve(...)

**Comment by Hit-And-Run-5677** [comment]
Also it is a crash as if I use await BunFile.text() in a Bun.serve(...) instead of new Response(Bun.file(...)), then the server breaks. And without "await" or then & catch, Bun gives a error called "pure virtual method called" before crashing. In the bun debugger, i created a Bun server with a await BunFile.text() in its fetch() method but when that line of code executes, immediately the debugger gets disabled.

**Comment by melroyvandenberg** [comment]
@Hit-And-Run-5677 move to Linux for now ;)

**Comment by mlorente** [comment]
same error with .text() and .json(), using the docker image "oven/bun:latest", try also using other linux distros as base image, but it still gives the same error as they say.

![image](https://github.com/oven-sh/bun/assets/6530189/5d5cbe86-43c8-4812-99bc-1252825feaff)

**Comment by melroy89** [comment]
I don't get this error with the `oven/bun` image on latest tag.. Using Docker version 24.0.6.

@mlorente are you able to debug the issue with: https://bun.sh/guides/runtime/web-debugger ?

```bash
root@55352a3e02a6:/data/local# cat index.ts 
console.log('start:');
const file = Bun.file("test.txt");
const text = await file.text();
console.log(text);
root@55352a3e02a6:/data/local# bun index.ts 
start:
hello world

root@55352a3e02a6:/data/local# uname -a
Linux 55352a3e02a6 6.2.0-33-generic #33~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Thu Sep  7 10:33:52 UTC 2 x86_64 GNU/Linux

root@55352a3e02a6:/data/local# cat /etc/*-release
PRETTY_NAME="Debian GNU/Linux 11 (bullseye)"
NAME="Debian GNU/Linux"
VERSION_ID="11"
VERSION="11 (bullseye)"
VERSION_CODENAME=bullseye
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
```

**Comment by siddacool** [comment]
I can confirm the same thing is happening with my WSL 1 setup as mentioned by  [SaifAshrafHelmy](https://github.com/SaifAshrafHelmy)

**Comment by crooowned** [comment]
Having the same, this needs a fix

**Comment by esjeon** [comment]
In my case, it turned out to be a linux capability issue. 

```
$ podman run -it --rm -v $PWD:/home/bun/app:z oven/bun run index.ts
start:
$ podman run --privileged -it --rm -v $PWD:/home/bun/app:z oven/bun run index.ts
start:
Hello World
```

So I ran `strace -f` on it, and found this:
```
[pid    71] io_uring_setup(1024, {flags=0, sq_thread_cpu=0, sq_thread_idle=1000 <unfinished ...> 
[pid    71] <... io_uring_setup resumed>}) = -1 ENOSYS (Function not implemented) 
```

Note that both docker and podman disabled `io_uring_*` by default due to security reasons.
* Docker: https://github.com/moby/moby/commit/891241e7e74d4aae6de5f6125574eb994f25e169
* Podman: https://github.com/containers/podman/issues/16796

**Comment by maliiiith** [comment]
Is this consistent across all the systems? I have a script parsing a JSON file, and it was built on Bun v1.0.3. It works fine locally (6.6.3-arch1-1) but on my Ubuntu 22 (5.15.0-89-generic) server, it no longer works even though it was working before. It just crashes.

**Comment by mnpenner** [comment]
I'm also seeing Bun quietly fail.

```ts
const schemaFile = Bun.file(`${__dirname}/../data/imagegather.yaml`)
const dbSchemaYaml = await schemaFile.text()
console.log(dbSchemaYaml)  // <-- never gets to this line
```

Using Bun `1.0.15+b3bdf22eb` on Windows/WSL1.

```ts
fs.readFileSync(`${__dirname}/../data/imagegather.yaml`, 'utf8')
``` 

works perfectly fine though, so I guess I'll keep using the node API.

**Comment by umstek** [comment]
``
